### PR TITLE
Update RELEASE_PROCESS and fix 'make deploy' for changes to juju 2

### PR DESCRIPTION
**Pull request by bac**

### CODE REVIEW

File: tests/deploy.py [hunk 1]
```diff
@@ -68,22 +68,22 @@ def juju_deploy(
     path = tempfile.mkdtemp()
     rsync(charm_source, path)
     args = ['deploy', '--series', series]
-    if service_name is None:
-        service_name = charm_name
+    if app_name is None:
+        app_name = charm_name
     if options is not None:
-        config_file = make_charm_config_file({service_name: options})
+        config_file = make_charm_config_file({app_name: options})
```
Comment (hatched) [inline tests/deploy.py]: Has this change been performed in this function? I don't see the change elsewhere.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/138/
Test FAILed.

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -45,9 +45,11 @@ Ensure the newly packaged charm deploys and behaves. Ensure you can create a
 new environment.
 
 ### Juju 1.25 ###
-     export JUJU_DEV_FEATURE_FLAGS=jes
```
Comment (jaycee) [inline RELEASE_PROCESS.md]: So we don't need to have the flag set anymore?

**Comment by jaycee** [comment]
Looks good, though I agree with @hatched's concern--doesn't look like that has been updated all the way through.

:+1: with that fixed, and with my assumption that juju-1 doesn't need the flag that you removed.

File: RELEASE_PROCESS.md (hunk 2 above)
Comment (bac) [inline RELEASE_PROCESS.md]: Nope. No longer support JES

File: tests/deploy.py (hunk 1 above)
Comment (bac) [inline tests/deploy.py]: That is just a variable. The variable name changed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/139/
Test PASSed.

**Comment by bac** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-charm-merge

**Pull request merged**